apps/mobile/src/features/send/send-form/hooks/use-send-form-stx.tsx

```diff
@@ -0,0 +1,52 @@
+import { useGenerateStxTokenTransferUnsignedTransaction } from '@/common/transactions/stacks-transactions.hooks';
+import { bytesToHex } from '@noble/hashes/utils';
+import BigNumber from 'bignumber.js';
+
+import { createMoneyFromDecimal } from '@leather.io/utils';
+
+import {
+  CreateCurrentSendRoute,
+  useSendSheetNavigation,
+  useSendSheetRoute,
+} from '../../send-form.utils';
+import { SendFormStxSchema } from '../schemas/send-form-stx.schema';
+
+export type CurrentRoute = CreateCurrentSendRoute<'send-form-stx'>;
+
+function parseSendFormValues(values: SendFormStxSchema) {
+  return {
+    amount: createMoneyFromDecimal(new BigNumber(values.amount), 'STX'),
+    fee: createMoneyFromDecimal(new BigNumber(values.fee), 'STX'),
+    memo: values.memo,
+    recipient: values.recipient,
+  };
+}
+
+export function useSendFormStx() {
+  const route = useSendSheetRoute<CurrentRoute>();
+  const navigation = useSendSheetNavigation<CurrentRoute>();
+
+  const generateTx = useGenerateStxTokenTransferUnsignedTransaction({
+    accountIndex: route.params.account.accountIndex,
+    fingerprint: route.params.account.fingerprint,
+  });
+
+  return {
+    onGoBack() {
+      navigation.navigate('send-select-asset', { account: route.params.account });
+    },
+    // Temporary logs until we can hook up to approver flow
+    async onSubmit(values: SendFormStxSchema) {
```

Comment:
Wonder if we could use a slightly more descriptive name for the action intended by the form, rather than form specific terminology.

`onInitiateSend`
`onUserInitTransfer`
`onSignTransaction` etc